--- xp-pen-pentablet-3.2.3.220323-1.x86_64/install.sh	2022-03-23 11:03:14.000000000 +0300
+++ src/xp-pen-pentablet-3.2.3.220323-1.x86_64/install.sh	2022-04-09 13:45:24.693404421 +0300
@@ -9,6 +9,20 @@
 echo $dirname
 cd "$dirname"
 
+
+
+rootDir="$1"
+if [ -n "${rootDir}" ]; then
+	if [ -d "${rootDir}" ]; then
+		rootDir="$(realpath "${rootDir}")"
+	else
+		echo "'${rootDir}' is not a directory"
+		exit 1
+	fi
+else
+	rootDir=""
+fi
+
 # close driver if it running
 AppName=pentablet
 AppDir=pentablet
@@ -37,8 +51,8 @@
 fi
 
 #Copy rule
-sysRuleDir="/lib/udev/rules.d"
-appRuleDir=./App$sysRuleDir
+sysRuleDir="${rootDir}/usr/lib/udev/rules.d"
+appRuleDir=./App/lib/udev/rules.d
 ruleName="10-xp-pen.rules"
 
 #echo "$appRuleDir/$ruleName"
@@ -55,8 +69,8 @@
 fi
 
 #install app
-sysAppDir="/usr/lib"
-appAppDir=./App$sysAppDir/$AppName
+sysAppDir="${rootDir}/usr/lib"
+appAppDir=./App/usr/lib/$AppName
 exeShell="pentablet.sh"
 
 #echo $sysAppDir
@@ -97,13 +111,13 @@
 fi
 
 # install shortcut
-sysDesktopDir=/usr/share/applications
-sysAppIconDir=/usr/share/icons
-sysAutoStartDir=/etc/xdg/autostart
-
-appDesktopDir=./App$sysDesktopDir
-appAppIconDir=./App$sysAppIconDir
-appAutoStartDir=./App$sysAutoStartDir
+sysDesktopDir="${rootDir}/usr/share/applications"
+sysAppIconDir="${rootDir}/usr/share/icons"
+sysAutoStartDir="${rootDir}/etc/xdg/autostart"
+
+appDesktopDir=./App/usr/share/applications
+appAppIconDir=./App/usr/share/icons
+appAutoStartDir=./App/etc/xdg/autostart
 
 appDesktopName=xp$AppName.desktop
 appIconName=$AppName.png
@@ -143,15 +157,15 @@
 	echo "Cannot find set auto start"
 fi
 
-absAppDir=/usr/lib/pentablet
+absAppDir=${rootDir}/usr/lib/pentablet
 absConfPath=$absAppDir/conf/xppen
 absLibPath=$absAppDir/lib
 absPlatPath=$absAppDir/platforms
 
 
-chmod +0777 /etc/xdg/autostart/xppentablet.desktop
-chmod +0777 /usr/share/applications/xppentablet.desktop
-chmod +0777 /usr/share/icons/pentablet.png
+chmod +0777 ${rootDir}/etc/xdg/autostart/xppentablet.desktop
+chmod +0777 ${rootDir}/usr/share/applications/xppentablet.desktop
+chmod +0777 ${rootDir}/usr/share/icons/pentablet.png
 
 chmod +0777 $absAppDir/pentablet
 chmod +0777 $absAppDir/pentablet.sh
@@ -184,10 +198,10 @@
 chmod +0666 $lockfile
 
 
-if [ -f "/etc/xdg/autostart/pentablet.desktop" ]; then
-	rm /etc/xdg/autostart/ugee-pentablet.desktop
+if [ -f "${rootDir}/etc/xdg/autostart/pentablet.desktop" ]; then
+	rm "${rootDir}/etc/xdg/autostart/ugee-pentablet.desktop"
 fi
 
 
-echo "Installation successful!"
+#echo "Installation successful!"
 echo "If you are installing for the first time, please use it after restart."
